src/compiler/checker.ts

```diff
@@ -32423,22 +32423,23 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             return getContextualCallSignature(type, node);
         }
         let signatureList: Signature[] | undefined;
-        const types = (type as UnionType).types;
+        const types = (type as UnionType).types
+            .map(type => ({ type, signature: getContextualCallSignature(type, node) }))
+            .filter(type => type.signature)

```

Comment:
Do you need to pass through `type` here? If I'm not mistaken, you could simplify this to:
```ts
const signatures = mapDefined((type as UnionType).types, t => getContextualCallSignature(t, node))
if (!signatures.length) {
  return;
}
const signatureList = signatures[0];
for (let i = 1; i < signatures.length; i++) {
  // loop body
}
```

Reply:
You are right, I was able to simplify, thank you.